### PR TITLE
More dep fixes

### PR DESCRIPTION
**Pull request by drcrallen**
At this point, here's the output of collision libraries:

```
druid-0.9.0-rc3-SNAPSHOT charlesallen$ find lib -iname *.jar | cut -d / -f 2 | sed -e 's/-[0-9]\.[0-9]/@/' | cut -f 1 -d @ | sort | uniq | xargs -I {} find extensions -name "*{}*.jar" | sort
extensions/druid-cloudfiles-extensions/guice-3.0.jar
extensions/druid-cloudfiles-extensions/guice-assistedinject-3.0.jar
extensions/druid-cloudfiles-extensions/guice-multibindings-3.0.jar
extensions/druid-cloudfiles-extensions/guice-multibindings-3.0.jar
extensions/druid-cloudfiles-extensions/guice-servlet-3.0.jar
extensions/druid-cloudfiles-extensions/guice-servlet-3.0.jar
extensions/druid-examples/RoaringBitmap-0.5.16.jar
extensions/druid-examples/activation-1.1.1.jar
extensions/druid-examples/aether-api-0.9.0.M2.jar
extensions/druid-examples/aether-connector-file-0.9.0.M2.jar
extensions/druid-examples/aether-connector-okhttp-0.0.9.jar
extensions/druid-examples/aether-connector-okhttp-0.0.9.jar
extensions/druid-examples/aether-impl-0.9.0.M2.jar
extensions/druid-examples/aether-spi-0.9.0.M2.jar
extensions/druid-examples/aether-util-0.9.0.M2.jar
extensions/druid-examples/airline-0.7.jar
extensions/druid-examples/annotations-2.0.3.jar
extensions/druid-examples/antlr4-runtime-4.0.jar
extensions/druid-examples/aopalliance-1.0.jar
extensions/druid-examples/aws-java-sdk-1.10.21.jar
extensions/druid-examples/aws-java-sdk-autoscaling-1.10.21.jar
extensions/druid-examples/aws-java-sdk-autoscaling-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudformation-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudformation-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudfront-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudfront-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudhsm-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudhsm-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudsearch-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudsearch-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudtrail-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudtrail-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudwatch-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudwatch-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudwatchmetrics-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudwatchmetrics-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cloudwatchmetrics-1.10.21.jar
extensions/druid-examples/aws-java-sdk-codecommit-1.10.21.jar
extensions/druid-examples/aws-java-sdk-codecommit-1.10.21.jar
extensions/druid-examples/aws-java-sdk-codedeploy-1.10.21.jar
extensions/druid-examples/aws-java-sdk-codedeploy-1.10.21.jar
extensions/druid-examples/aws-java-sdk-codepipeline-1.10.21.jar
extensions/druid-examples/aws-java-sdk-codepipeline-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cognitoidentity-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cognitoidentity-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cognitosync-1.10.21.jar
extensions/druid-examples/aws-java-sdk-cognitosync-1.10.21.jar
extensions/druid-examples/aws-java-sdk-config-1.10.21.jar
extensions/druid-examples/aws-java-sdk-config-1.10.21.jar
extensions/druid-examples/aws-java-sdk-core-1.10.21.jar
extensions/druid-examples/aws-java-sdk-core-1.10.21.jar
extensions/druid-examples/aws-java-sdk-datapipeline-1.10.21.jar
extensions/druid-examples/aws-java-sdk-datapipeline-1.10.21.jar
extensions/druid-examples/aws-java-sdk-devicefarm-1.10.21.jar
extensions/druid-examples/aws-java-sdk-devicefarm-1.10.21.jar
extensions/druid-examples/aws-java-sdk-directconnect-1.10.21.jar
extensions/druid-examples/aws-java-sdk-directconnect-1.10.21.jar
extensions/druid-examples/aws-java-sdk-directory-1.10.21.jar
extensions/druid-examples/aws-java-sdk-directory-1.10.21.jar
extensions/druid-examples/aws-java-sdk-dynamodb-1.10.21.jar
extensions/druid-examples/aws-java-sdk-dynamodb-1.10.21.jar
extensions/druid-examples/aws-java-sdk-ec2-1.10.21.jar
extensions/druid-examples/aws-java-sdk-ec2-1.10.21.jar
extensions/druid-examples/aws-java-sdk-ecs-1.10.21.jar
extensions/druid-examples/aws-java-sdk-ecs-1.10.21.jar
extensions/druid-examples/aws-java-sdk-efs-1.10.21.jar
extensions/druid-examples/aws-java-sdk-efs-1.10.21.jar
extensions/druid-examples/aws-java-sdk-elasticache-1.10.21.jar
extensions/druid-examples/aws-java-sdk-elasticache-1.10.21.jar
extensions/druid-examples/aws-java-sdk-elasticbeanstalk-1.10.21.jar
extensions/druid-examples/aws-java-sdk-elasticbeanstalk-1.10.21.jar
extensions/druid-examples/aws-java-sdk-elasticloadbalancing-1.10.21.jar
extensions/druid-examples/aws-java-sdk-elasticloadbalancing-1.10.21.jar
extensions/druid-examples/aws-java-sdk-elastictranscoder-1.10.21.jar
extensions/druid-examples/aws-java-sdk-elastictranscoder-1.10.21.jar
extensions/druid-examples/aws-java-sdk-emr-1.10.21.jar
extensions/druid-examples/aws-java-sdk-emr-1.10.21.jar
extensions/druid-examples/aws-java-sdk-glacier-1.10.21.jar
extensions/druid-examples/aws-java-sdk-glacier-1.10.21.jar
extensions/druid-examples/aws-java-sdk-iam-1.10.21.jar
extensions/druid-examples/aws-java-sdk-iam-1.10.21.jar
extensions/druid-examples/aws-java-sdk-importexport-1.10.21.jar
extensions/druid-examples/aws-java-sdk-importexport-1.10.21.jar
extensions/druid-examples/aws-java-sdk-kinesis-1.10.21.jar
extensions/druid-examples/aws-java-sdk-kinesis-1.10.21.jar
extensions/druid-examples/aws-java-sdk-kms-1.10.21.jar
extensions/druid-examples/aws-java-sdk-kms-1.10.21.jar
extensions/druid-examples/aws-java-sdk-lambda-1.10.21.jar
extensions/druid-examples/aws-java-sdk-lambda-1.10.21.jar
extensions/druid-examples/aws-java-sdk-logs-1.10.21.jar
extensions/druid-examples/aws-java-sdk-logs-1.10.21.jar
extensions/druid-examples/aws-java-sdk-machinelearning-1.10.21.jar
extensions/druid-examples/aws-java-sdk-machinelearning-1.10.21.jar
extensions/druid-examples/aws-java-sdk-opsworks-1.10.21.jar
extensions/druid-examples/aws-java-sdk-opsworks-1.10.21.jar
extensions/druid-examples/aws-java-sdk-rds-1.10.21.jar
extensions/druid-examples/aws-java-sdk-rds-1.10.21.jar
extensions/druid-examples/aws-java-sdk-redshift-1.10.21.jar
extensions/druid-examples/aws-java-sdk-redshift-1.10.21.jar
extensions/druid-examples/aws-java-sdk-route53-1.10.21.jar
extensions/druid-examples/aws-java-sdk-route53-1.10.21.jar
extensions/druid-examples/aws-java-sdk-s3-1.10.21.jar
extensions/druid-examples/aws-java-sdk-s3-1.10.21.jar
extensions/druid-examples/aws-java-sdk-ses-1.10.21.jar
extensions/druid-examples/aws-java-sdk-ses-1.10.21.jar
extensions/druid-examples/aws-java-sdk-simpledb-1.10.21.jar
extensions/druid-examples/aws-java-sdk-simpledb-1.10.21.jar
extensions/druid-examples/aws-java-sdk-simpleworkflow-1.10.21.jar
extensions/druid-examples/aws-java-sdk-simpleworkflow-1.10.21.jar
extensions/druid-examples/aws-java-sdk-sns-1.10.21.jar
extensions/druid-examples/aws-java-sdk-sns-1.10.21.jar
extensions/druid-examples/aws-java-sdk-sqs-1.10.21.jar
extensions/druid-examples/aws-java-sdk-sqs-1.10.21.jar
extensions/druid-examples/aws-java-sdk-ssm-1.10.21.jar
extensions/druid-examples/aws-java-sdk-ssm-1.10.21.jar
extensions/druid-examples/aws-java-sdk-storagegateway-1.10.21.jar
extensions/druid-examples/aws-java-sdk-storagegateway-1.10.21.jar
extensions/druid-examples/aws-java-sdk-sts-1.10.21.jar
extensions/druid-examples/aws-java-sdk-sts-1.10.21.jar
extensions/druid-examples/aws-java-sdk-support-1.10.21.jar
extensions/druid-examples/aws-java-sdk-support-1.10.21.jar
extensions/druid-examples/aws-java-sdk-swf-libraries-1.10.21.jar
extensions/druid-examples/aws-java-sdk-swf-libraries-1.10.21.jar
extensions/druid-examples/aws-java-sdk-workspaces-1.10.21.jar
extensions/druid-examples/aws-java-sdk-workspaces-1.10.21.jar
extensions/druid-examples/base64-2.3.8.jar
extensions/druid-examples/bcprov-jdk15on-1.52.jar
extensions/druid-examples/bytebuffer-collections-0.2.4.jar
extensions/druid-examples/classmate-1.0.0.jar
extensions/druid-examples/commons-cli-1.2.jar
extensions/druid-examples/commons-codec-1.7.jar
extensions/druid-examples/commons-dbcp2-2.0.1.jar
extensions/druid-examples/commons-io-2.4.jar
extensions/druid-examples/commons-lang-2.6.jar
extensions/druid-examples/commons-logging-1.1.1.jar
extensions/druid-examples/commons-pool-1.6.jar
extensions/druid-examples/commons-pool2-2.2.jar
extensions/druid-examples/commons-pool2-2.2.jar
extensions/druid-examples/compress-lzf-1.0.3.jar
extensions/druid-examples/config-magic-0.9.jar
extensions/druid-examples/curator-client-2.9.1.jar
extensions/druid-examples/curator-framework-2.9.1.jar
extensions/druid-examples/curator-recipes-2.9.1.jar
extensions/druid-examples/curator-x-discovery-2.9.1.jar
extensions/druid-examples/disruptor-3.3.0.jar
extensions/druid-examples/druid-api-0.3.16.jar
extensions/druid-examples/druid-aws-common-0.9.0-rc3-SNAPSHOT.jar
extensions/druid-examples/druid-common-0.9.0-rc3-SNAPSHOT.jar
extensions/druid-examples/druid-console-0.0.2.jar
extensions/druid-examples/druid-processing-0.9.0-rc3-SNAPSHOT.jar
extensions/druid-examples/druid-server-0.9.0-rc3-SNAPSHOT.jar
extensions/druid-examples/emitter-0.3.6.jar
extensions/druid-examples/extendedset-1.3.9.jar
extensions/druid-examples/geoip2-0.4.0.jar
extensions/druid-examples/google-http-client-1.15.0-rc.jar
extensions/druid-examples/google-http-client-1.15.0-rc.jar
extensions/druid-examples/google-http-client-jackson2-1.15.0-rc.jar
extensions/druid-examples/google-http-client-jackson2-1.15.0-rc.jar
extensions/druid-examples/google-http-client-jackson2-1.15.0-rc.jar
extensions/druid-examples/guice-4.0-beta.jar
extensions/druid-examples/guice-multibindings-4.0-beta.jar
extensions/druid-examples/guice-multibindings-4.0-beta.jar
extensions/druid-examples/guice-servlet-4.0-beta.jar
extensions/druid-examples/guice-servlet-4.0-beta.jar
extensions/druid-examples/hibernate-validator-5.1.3.Final.jar
extensions/druid-examples/http-client-1.0.4.jar
extensions/druid-examples/httpclient-4.5.1.jar
extensions/druid-examples/httpcore-4.4.3.jar
extensions/druid-examples/icu4j-4.8.1.jar
extensions/druid-examples/irc-api-1.0-0014.jar
extensions/druid-examples/jackson-annotations-2.4.6.jar
extensions/druid-examples/jackson-annotations-2.4.6.jar
extensions/druid-examples/jackson-core-2.4.6.jar
extensions/druid-examples/jackson-core-asl-1.9.13.jar
extensions/druid-examples/jackson-core-asl-1.9.13.jar
extensions/druid-examples/jackson-databind-2.4.6.jar
extensions/druid-examples/jackson-dataformat-smile-2.4.6.jar
extensions/druid-examples/jackson-datatype-guava-2.4.6.jar
extensions/druid-examples/jackson-datatype-joda-2.4.6.jar
extensions/druid-examples/jackson-jaxrs-base-2.4.6.jar
extensions/druid-examples/jackson-jaxrs-json-provider-2.4.6.jar
extensions/druid-examples/jackson-jaxrs-smile-provider-2.4.6.jar
extensions/druid-examples/jackson-mapper-asl-1.9.13.jar
extensions/druid-examples/jackson-module-jaxb-annotations-2.4.6.jar
extensions/druid-examples/jackson-module-jaxb-annotations-2.4.6.jar
extensions/druid-examples/java-util-0.27.7.jar
extensions/druid-examples/java-xmlbuilder-1.1.jar
extensions/druid-examples/javax.el-3.0.0.jar
extensions/druid-examples/javax.el-api-3.0.0.jar
extensions/druid-examples/javax.el-api-3.0.0.jar
extensions/druid-examples/javax.servlet-api-3.1.0.jar
extensions/druid-examples/jboss-logging-3.1.3.GA.jar
extensions/druid-examples/jcl-over-slf4j-1.7.12.jar
extensions/druid-examples/jdbi-2.63.1.jar
extensions/druid-examples/jersey-core-1.19.jar
extensions/druid-examples/jersey-guice-1.19.jar
extensions/druid-examples/jersey-guice-1.19.jar
extensions/druid-examples/jersey-server-1.19.jar
extensions/druid-examples/jersey-servlet-1.19.jar
extensions/druid-examples/jets3t-0.9.4.jar
extensions/druid-examples/jetty-client-9.2.5.v20141112.jar
extensions/druid-examples/jetty-continuation-9.2.5.v20141112.jar
extensions/druid-examples/jetty-http-9.2.5.v20141112.jar
extensions/druid-examples/jetty-io-9.2.5.v20141112.jar
extensions/druid-examples/jetty-proxy-9.2.5.v20141112.jar
extensions/druid-examples/jetty-security-9.2.5.v20141112.jar
extensions/druid-examples/jetty-server-9.2.5.v20141112.jar
extensions/druid-examples/jetty-servlet-9.2.5.v20141112.jar
extensions/druid-examples/jetty-servlets-9.2.5.v20141112.jar
extensions/druid-examples/jetty-servlets-9.2.5.v20141112.jar
extensions/druid-examples/jetty-util-9.2.5.v20141112.jar
extensions/druid-examples/jline-0.9.94.jar
extensions/druid-examples/joda-time-2.8.2.jar
extensions/druid-examples/json-path-2.1.0.jar
extensions/druid-examples/jsr305-2.0.1.jar
extensions/druid-examples/jsr311-api-1.1.1.jar
extensions/druid-examples/log4j-1.2-api-2.5.jar
extensions/druid-examples/log4j-api-2.5.jar
extensions/druid-examples/log4j-api-2.5.jar
extensions/druid-examples/log4j-core-2.5.jar
extensions/druid-examples/log4j-core-2.5.jar
extensions/druid-examples/log4j-jul-2.5.jar
extensions/druid-examples/log4j-jul-2.5.jar
extensions/druid-examples/log4j-slf4j-impl-2.5.jar
extensions/druid-examples/log4j-slf4j-impl-2.5.jar
extensions/druid-examples/lz4-1.3.0.jar
extensions/druid-examples/mapdb-1.0.8.jar
extensions/druid-examples/maven-aether-provider-3.1.1.jar
extensions/druid-examples/maven-model-3.1.1.jar
extensions/druid-examples/maven-model-builder-3.1.1.jar
extensions/druid-examples/maven-model-builder-3.1.1.jar
extensions/druid-examples/maven-repository-metadata-3.1.1.jar
extensions/druid-examples/maven-settings-3.1.1.jar
extensions/druid-examples/maven-settings-builder-3.1.1.jar
extensions/druid-examples/maven-settings-builder-3.1.1.jar
extensions/druid-examples/maxminddb-0.2.0.jar
extensions/druid-examples/netty-3.10.4.Final.jar
extensions/druid-examples/okhttp-1.0.2.jar
extensions/druid-examples/opencsv-2.3.jar
extensions/druid-examples/org.abego.treelayout.core-1.0.1.jar
extensions/druid-examples/plexus-interpolation-1.19.jar
extensions/druid-examples/plexus-utils-3.0.15.jar
extensions/druid-examples/protobuf-java-2.5.0.jar
extensions/druid-examples/rhino-1.7R5.jar
extensions/druid-examples/server-metrics-0.2.8.jar
extensions/druid-examples/slf4j-api-1.7.12.jar
extensions/druid-examples/spymemcached-2.11.7.jar
extensions/druid-examples/tesla-aether-0.0.5.jar
extensions/druid-examples/validation-api-1.1.0.Final.jar
extensions/druid-examples/wagon-provider-api-2.4.jar
extensions/druid-examples/xpp3-1.1.4c.jar
extensions/druid-examples/zookeeper-3.4.6.jar
```

### CODE REVIEW

File: services/src/main/java/io/druid/cli/PullDependencies.java [hunk 1]
```diff
@@ -94,6 +93,14 @@
       // Different tasks which are classloader sensitive attempt to maintain a sane order for loading libraries in the
       // classloader, but it is always possible that something didn't load in the right order. Also we don't want to be
       // throwing around a ton of jars we don't need to.
+      //
+      // Here is a list of dependencies extensions should probably exclude.
+      //
+      // Conflicts can be discovered using the following command on the distribution tarball:
+      //    `find lib -iname *.jar | cut -d / -f 2 | sed -e 's/-[0-9]\.[0-9]/@/' | cut -f 1 -d @ | sort | uniq | xargs -I {} find extensions -name "*{}*.jar" | sort`
```
Comment (nishantmonu51) [inline services/src/main/java/io/druid/cli/PullDependencies.java]: probably this should be moved to docs for extending druid? 
http://druid.io/docs/latest/development/modules.html

Comment (drcrallen) [inline services/src/main/java/io/druid/cli/PullDependencies.java]: Fixed

File: extensions/hdfs-storage/pom.xml [hunk 2]
```diff
@@ -94,6 +87,48 @@
           <version>${hadoop.compile.version}</version>
           <classifier>tests</classifier>
           <scope>test</scope>
+          <exclusions>
```
Comment (xvrl) [inline extensions/hdfs-storage/pom.xml]: this is in test scope, so it shouldn't affect anything at runtime

File: extensions/s3-extensions/pom.xml [hunk 3]
```diff
@@ -48,24 +48,6 @@
         <dependency>
             <groupId>net.java.dev.jets3t</groupId>
             <artifactId>jets3t</artifactId>
-            <scope>provided</scope>
```
Comment (xvrl) [inline extensions/s3-extensions/pom.xml]: any reason jets3t is not in "provided" scope anymore?

Comment (drcrallen) [inline extensions/s3-extensions/pom.xml]: mistake, fixed.

File: extensions/hdfs-storage/pom.xml (hunk 2 above)
Comment (drcrallen) [inline extensions/hdfs-storage/pom.xml]: removed for clarity of file

**Comment by nishantmonu51** [comment]
:+1:

File: extensions/cassandra-storage/pom.xml [hunk 4]
```diff
@@ -34,15 +34,88 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>com.ning</groupId>
+                    <artifactId>compress-lzf</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>commons-lang</groupId>
+                    <artifactId>commons-lang</artifactId>
+                </exclusion>
+                <!-- Coppied from hadoop exclusions -->
```
Comment (xvrl) [inline extensions/cassandra-storage/pom.xml]: typo

File: indexing-hadoop/pom.xml [hunk 5]
```diff
@@ -53,19 +53,23 @@
         <dependency>
             <groupId>net.java.dev.jets3t</groupId>
             <artifactId>jets3t</artifactId>
+	    <scope>test</scope>
```
Comment (xvrl) [inline indexing-hadoop/pom.xml]: indentation is off

File: aws-common/pom.xml [hunk 6]
```diff
@@ -39,6 +39,19 @@
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk</artifactId>
         </dependency>
+        <!--
+        There's a bug in
+        https://github.com/apache/httpclient/blob/4.5.2/httpclient/src/main/java-deprecated/org/apache/http/impl/client/AbstractHttpClient.java#L332
+        Where httpclient does not care what your context classloader is when looking for
```
Comment (xvrl) [inline aws-common/pom.xml]: when looking for ...?

**Pull request merged**